test/tools/runner/throw_rejections.cjs

```diff
@@ -0,0 +1,8 @@
+/* eslint-disable no-console */
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+const process = require('process');
+
+process.on('unhandledRejection', (error, promise) => {
+  console.log('promise:', promise, 'unhandledRejection:', error);
```

Comment:
The print here helps tell the difference between a rejection and a normal uncaught error when you scroll back up to where it was printed. Happy to drop though.

Reply:
Evergreen interleaves logs so that stderr and stdout are mixed - errors are printed to stderr and logs to stdout.  That means that there's no guarantee that the print statement would immediately precede the log from Node

Maybe we could wrap the error in a new error?